fuzz/cpio_rdr.c

```diff
@@ -365,15 +375,16 @@ const char *cpio_readentry(CPIO *cpio, size_t *datasize)
     filenamesize = cpio->reader->filenamesize(header);
     *datasize = cpio->datasize = cpio->reader->datasize(header);
 
-    cpio->rel_dataoffset =
-        ((cpio->rel_filenameoffset + filenamesize + cpio->reader->align - 1)
+    cpio->r_dataoffset =
+        ((cpio->r_filenameoffset + filenamesize + cpio->reader->align - 1)
          / cpio->reader->align)
         * cpio->reader->align;
     cpio->datasize_aligned =
         ((*datasize + cpio->reader->align - 1) / cpio->reader->align)
         * cpio->reader->align;
 
-    if (fread(pathname, cpio->rel_dataoffset - cpio->rel_filenameoffset, 1,
+    if (fread(pathname,
+              (size_t)cpio->r_dataoffset - (size_t)cpio->r_filenameoffset, 1,
```

Comment:
check for underflow ?

Reply:
I don't see the need.  `r_dataoffset` is calculated 9 lines up as `r_filenameoffset` plus the file name size.  True, I could have used `filenamesize` directly, but then I would have to fix alignment another way.  This calculation includes the needed alignment.